File: android/app/src/main/java/com/smartexpense/android/domain/repository/AuthRepository.java
package com.smartexpense.android.domain.repository;

import com.smartexpense.android.data.remote.dto.request.LoginRequestDto;
import com.smartexpense.android.data.remote.dto.request.RegisterRequestDto;
import com.smartexpense.android.data.remote.dto.response.AuthResponseDto;

public interface AuthRepository {
    void login(LoginRequestDto request, ResultCallback<AuthResponseDto> callback);
    void register(RegisterRequestDto request, ResultCallback<AuthResponseDto> callback);
}

File: android/app/src/main/kotlin/com/smartexpense/android/domain/repository/AuthRepository.kt
package com.smartexpense.android.domain.repository

import com.smartexpense.android.data.remote.dto.request.LoginRequestDto
import com.smartexpense.android.data.remote.dto.request.RegisterRequestDto
import com.smartexpense.android.data.remote.dto.response.AuthResponseDto

interface AuthRepository {
    suspend fun login(request: LoginRequestDto): Result<AuthResponseDto>
    suspend fun register(request: RegisterRequestDto): Result<AuthResponseDto>
}
